Lib/glyphsLib/builder/paths.py

```diff
@@ -40,11 +40,24 @@ def to_ufo_paths(self, ufo_glyph, layer):
             # In Glyphs.app, the starting node of a closed contour is always
             # stored at the end of the nodes list.
             nodes.insert(0, nodes.pop())
+
         for node in nodes:
             node_type = _to_ufo_node_type(node.type)
+            node_name = node.userData.get("name")
+            if node_name is not None:
+                # Without a proxy, one could `node.userData.pop("name", None)` instead.
+                del node.userData["name"]
```

Comment:
I still don't get why you're changing stuff in the Glyphs userData while in a "to_ufo..." function?

Reply:
Because we write the Glyphs node user data to a glyph lib key. Glyph stores the node name in a node's user data, but the UFO spec bolts it onto the actual point, so we have to extract the name from user data and then delete it from there, so we don't write the node name to actual point AND lib key.

Reply:
Hrm. this does change the input file though. Grrr.

Reply:
The code now copies stuff instead of changing the input.